docs/1.concepts/storage/decentralized-storage.md

```diff
@@ -12,6 +12,18 @@ sidebar_label: Alternative Solutions
 
 ---
 
+## On-Chain Storage Constraints 
+
+For storing data on-chain it's important to keep in mind the following: 
+
+- You can store an unlimited amount of file, but will cost you 1N per 100KB
+- There is a 4mb limit on how much you can upload at once 
+
+
+Let's say for example, someone wants to put an NFT purely on-chain (rather than IPFS or some other decentralized storage solution as mentioned below) you'll have almost an unlimited amount of storage but will have to pay 1 $NEAR per 100kb of storage used (see [Storage Staking](https://docs.near.org/concepts/storage/storage-staking))
```

Comment:
```suggestion
For example, if you want to store an NFT purely on-chain (rather than using IPFS or some other decentralized storage solution as mentioned below) you'll have almost an unlimited amount of storage but will have to pay 1 $NEAR per 100 KB of storage used (see [Storage Staking](https://docs.near.org/concepts/storage/storage-staking))
```